src/components/TicketFeedContainer/TicketFeedContainer.tsx

```diff
@@ -1,26 +1,49 @@
-import { Box } from "@mui/material"
-import { InsightDataInterface } from "../../../interfaces/InsightData"
+import {Box, Button} from "@mui/material"
 import InsightCard from "../InsightCard/InsightCard"
+import {Insight} from "@interfaces/insight";
+import {FiledRequest} from "@interfaces/filedRequest";
+import {useState} from "react";
+import FiledRequestCard from "@components/FiledRequestCard/FiledRequestCard";
 
-interface TicketFeedContainerInterface { 
-    insightData: InsightDataInterface[]
+interface TicketFeedContainerInterface {
+    filedRequestData: {[id: string]: FiledRequest},
+    deleteFiledRequestHandlerGenerator: Function,
+    insightData: {[id: string]: Insight},
+    deleteInsightHandlerGenerator: Function
 }
 
-const TicketFeedContainer = ({ insightData }: TicketFeedContainerInterface) => {
+type DisplayType = "Requests" | "Insights"
+
+const TicketFeedContainer = ({   filedRequestData,
+                                 deleteFiledRequestHandlerGenerator,
+                                 insightData,
+                                 deleteInsightHandlerGenerator }: TicketFeedContainerInterface) => {
+    const [currentDisplay, setCurrentDisplay] = useState<DisplayType>("Requests");
+
+    console.log(filedRequestData)
+    console.log(Object.entries(filedRequestData))
     return (
-        <Box sx={{
-            border: '1px #000 solid', 
-            background: '#fff', 
-            width: '100vw', 
-            height:'100vh',
-            marginTop:'100px'
-            }}>
+        <Box>
             <p style={{color: "#000", textAlign: 'end', padding:'1rem 2rem'}}>Sort by Recommended</p>
+            {/**TODO: Current button placement is very cursed, need to adjust**/}
+            <Button  onClick={() => {setCurrentDisplay("Requests");} }>
+                Show Requests
+            </Button>
+            <Button  onClick={() => {setCurrentDisplay("Insights");} }>
+                Show Insights
+            </Button>
+            {currentDisplay == "Requests" &&
```

Comment:
Since you're reusing the Box here, you can put a ternary inside the Box `currentDisplay === "Requests" ? <></> : <></>` , the onDelete might have to be `() => deleteFiledRequestHandlerGenerator(filedRequestId)`

Reply:
Done